aixplain/modules/model.py

```
@@ -249,3 +252,36 @@ def run_async(self, data: Union[Text, Dict], name: Text = "model_process", param
             if resp is not None:
                 response["error"] = msg
         return response
+    
+    def check_finetune_status(self):
+        """Check the status of the FineTune model.
+
+        Raises:
+            Exception: If the 'TEAM_API_KEY' is not provided.
+
+        Returns:
+            str: The status of the FineTune model.
+        """
+        if self.api_key == "":
+            raise Exception(
+                "A 'TEAM_API_KEY' is required to run a model. For help, please refer to the documentation (https://github.com/aixplain/aixplain#api-key-setup)"
+            )
+        headers = {"x-api-key": self.api_key, "Content-Type": "application/json"}
+        try:
+            url = urljoin(self.backend_url, f"sdk/models/{self.id}")
+            logging.info(f"Start service for GET Metric  - {url} - {headers}")
+            r = _request_with_retry("get", url, headers=headers)
+            resp = r.json()
+            status = resp["status"]
+            logging.info(f"Model Check FineTune: Model {self.id} - status {status}.")
+            return status
+        except Exception as e:
+            if resp is not None and "statusCode" in resp:
+                status_code = resp["statusCode"]
+                message = resp["message"]
+                message = f"Model Check FineTune: Status {status_code} - {message}"
+            else:
+                message = "Model Check FineTune: Unspecified Error"
+            logging.error(message)
+            raise Exception(f"{message}")
```

Why do we want to suppress the exception and re-raise it?